tests/integration.rs

```diff
@@ -376,11 +366,23 @@ fn listxattr(#[case] harness: Harness) {
 	let xattrs = file.list_xattr().unwrap().collect::<Vec<_>>();
 	let expected = [OsStr::new("user.test")];
 	assert_eq!(xattrs, expected);
+}
 
-	freebsd! {
-		let num = unsafe { libc::extattr_list_fd(file.as_raw_fd(), EXTATTR_NAMESPACE_USER, std::ptr::null_mut(), 0) };
-		assert_eq!(num, 5); // strlen("test\0")
-	}
+#[cfg(target_os = "freebsd")]
+#[apply(all_images)]
+fn listxattr_freebsd(#[case] harness: Harness) {
```

Comment:
The purpose of the test is to do a size-only query.  It's restricted to FreeBSD just because the most convenient API to use is FreeBSD-only.  So I suggest changing the name.
```suggestion
fn listxattr_size(#[case] harness: Harness) {
```